src/lib/operators/join_mpsm.cpp

```diff
@@ -38,24 +38,26 @@ STRONG_TYPEDEF(uint32_t, ClusterID);
 namespace opossum {
 JoinMPSM::JoinMPSM(const std::shared_ptr<const AbstractOperator>& left,
                    const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
-                   const std::pair<ColumnID, ColumnID>& column_ids, const PredicateCondition op)
-    : AbstractJoinOperator(OperatorType::JoinMPSM, left, right, mode, column_ids, op) {
+                   const OperatorJoinPredicate& primary_predicate)
+    : AbstractJoinOperator(OperatorType::JoinMPSM, left, right, mode, primary_predicate, {}) {
   // Validate the parameters
   DebugAssert(mode != JoinMode::Cross, "This operator does not support cross joins.");
   DebugAssert(left != nullptr, "The left input operator is null.");
   DebugAssert(right != nullptr, "The right input operator is null.");
-  DebugAssert(op == PredicateCondition::Equals, "Only Equi joins are supported by MPSM join.");
+  DebugAssert(primary_predicate.predicate_condition == PredicateCondition::Equals,
```

Comment:
Does the MPSM join really support multiple predicates? 

If not, please assert.

Reply:
JoinMPSM doesn't even take secondary predicates as a constructor arg so I think it is sufficiently ensured that it isn't constructed with such.